util/unzipAar.gradle

```diff
@@ -0,0 +1,62 @@
+/**
+ * Adds an `unzipAarToJar` task to any project that inculdes this file.
+ * The task will collect the compile dependencies of the project and it's local project
+ * dependencies and unzip the classes.jar from aars to `ext.aarJarDir`
+ */
+
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.util.zip.ZipFile
+
+static def extractEntryFromZip(archive, entryPath, outputPath) {
+  if (!archive.exists()) {
+    throw new GradleException("archive $archive not found")
+  }
+
+  def zip = new ZipFile(archive)
+  zip.entries().each {
+    if (it.name == entryPath) {
+      def path = Paths.get(outputPath)
+      if (!Files.exists(path)) {
+        Files.createDirectories(path.getParent())
+        Files.copy(zip.getInputStream(it), path)
+      }
+    }
+  }
+  zip.close()
+}
+
+static Set binaryCompileDependencies(Project project) {
+  if(project.configurations.findByName('compile')) {
+    def compileDependencies = project.configurations.compile
+
+    return project.configurations.compile.allDependencies.findAll {
+      it instanceof ProjectDependency
+    }.collect {
+      binaryCompileDependencies(it.dependencyProject)
+    }.inject(compileDependencies.files) { left, right ->
+      left + right
+    }
+  } else {
+    return [] as Set
+  }
+}
+
+if(tasks.findByName('unzipAarToJar')) {
+  println 'unzipAarToJar already added, won\'t try to add it again'
```

Comment:
Not sure why we need to know that (it pollutes the log imho).

Reply:
fair enough